src/node_file.cc

```diff
@@ -885,6 +885,63 @@ static void WriteBuffer(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+// Wrapper for writev(2).
+//
+// bytesWritten = writev(fd, chunks, position, callback)
+// 0 fd        integer. file descriptor
+// 1 chunks    array of buffers to write
+// 2 position  if integer, position to write at in the file.
+//             if null, write from the current position
+static void WriteBuffers(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  CHECK(args[0]->IsInt32());
+  CHECK(args[1]->IsArray());
+
+  int fd = args[0]->Int32Value();
+  Local<Array> chunks = args[1].As<Array>();
+  int64_t pos = GET_OFFSET(args[2]);
+  Local<Value> req = args[3];
+
+  uint32_t chunkCount = chunks->Length();
+
+  uv_buf_t s_iovs[1024];       // stack allocated
+  uv_buf_t* h_iovs = nullptr;  // heap allocated (if it won't fit on the stack)
+  uv_buf_t* iovs;
+
+  if (chunkCount > 1024) {
+    h_iovs = new uv_buf_t[chunkCount];
+    iovs = h_iovs;
+  } else {
+    iovs = s_iovs;
+  }
```

Comment:
You can simplify this by just having s_iovs and iovs, and later comparing if they are the same.

~~~~
if (chunkCount > ARRAY_SIZE(s_iovs)) {
    iovs = new uv_buf_t[chunkCount];
    // TODO: check if allocation failed, I guess
} else {
    iovs = s_iovs;
}

....

if (iovs != s_iovs) {
    delete[] iovs;
}
~~~~